### ISSUE TITLE
Scripts location when developing rules SHOULD BE scripts, not rules.

### ISSUE THREAD

**Issue by timo12357** [openhab/openhab-webui#1921]
## The problem
In OH3 using the rules menu option I have come across a counterintuitive UI flow:

When developing a rule with “Run Script” it does not allow you to select a Script from those developed under the “Scripts” menu option, instead it forces you to write a new script each time and appears to store it locally within the rule. One would have expected all scripts to be stored in one common location rather than some being stored under the “Scripts” menu option and others being stored within individual rules. 

Further, the scripts developed under the "Scripts" menu option are available under "Other rules", which requires quite a long logical leap to figure out intuitively:

![image](https://github.com/openhab/openhab-webui/assets/11517762/2ea44759-7530-40a0-aca4-f0b0dba44231)

## Your suggestion
The UI flow would be intuitive, if and when the scripts developed under the "Scripts" menu would be available to be selected for the rule under development from the "Run scripts" menu option.

## Your environment

runtimeInfo:
  version: 3.4.4
  buildString: Release Build
locale: fi-FI
systemInfo:
  configFolder: /etc/openhab
  userdataFolder: /var/lib/openhab
  logFolder: /var/log/openhab
  javaVersion: 11.0.9.1
  javaVendor: Azul Systems, Inc.
  javaVendorVersion: Zulu11.43+100-CA
  osName: Linux
  osVersion: 5.15.93-sunxi64
  osArchitecture: aarch64
  availableProcessors: 4
  freeMemory: 79296720
  totalMemory: 451936256
  startLevel: 100
bindings:
  - astro
  - exec
  - homeconnect
  - mqtt
  - shelly
  - solarforecast
  - telegram
  - tuya
  - zigbee
clientInfo:
  device:
    ios: false
    android: false
    androidChrome: false
    desktop: true
    iphone: false
    ipod: false
    ipad: false
    edge: false
    ie: false
    firefox: true
    macos: false
    windows: false
    cordova: false
    phonegap: false
    electron: false
    nwjs: false
    webView: false
    webview: false
    standalone: false
    pixelRatio: 1
    prefersColorScheme: light
  isSecureContext: true
  locationbarVisible: true
  menubarVisible: true
  navigator:
    cookieEnabled: true
    deviceMemory: N/A
    hardwareConcurrency: 8
    language: en-US
    languages:
      - en-US
      - en
    onLine: true
    platform: Linux x86_64
  screen:
    width: 1920
    height: 1080
    colorDepth: 24
  support:
    touch: false
    pointerEvents: true
    observer: true
    passiveListener: true
    gestures: false
    intersectionObserver: true
  themeOptions:
    dark: light
    filled: true
    pageTransitionAnimation: default
    bars: filled
    homeNavbar: default
    homeBackground: default
    expandableCardAnimation: default
  userAgent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0) Gecko/20100101
    Firefox/113.0
timestamp: 2023-06-09T15:58:23.046Z

```

## Additional information
OpenHAB is great, thank you for your efforts in making this platform so good!
<!--
Provide any information not pertinent to the above sections that you'd still like to share.
-->

**Comment by rkoshak** [comment]
I'm not sure I agree with the suggested fix, because Script Actions and Script Conditions (which you create under the `Run Script` menu) are fundamentally different from the Scripts under the Settings -> Scripts tab in MainUI. Treating them like they are the same will, IMO, *increase* the confusion, not ease confusion.

I think perhaps some changes in terminology, which preserves and highlights these differences might be a better approach. I'm not sure what would be best.

Maybe change "Run a Script" to read "Create a Script" would help clear things up on the one side. That should be rather benign of a change.

Then maybe changing the "Other Rules" to read "Other Rules, Scripts, Schedules and Scenes" (do we even need to mention Schedules? I think those appear under Rules too, or are Schedule tagged rules only shown under Schedules)?

Were I king I'd change the name of Scripts to something else , maybe "Rule Fragment". But that might be too disruptive.

**Comment by ghys** [comment]
That's been on my mind for years, there's just too many things we refer as "scripts".

1. Files in the `$OPENHAB_CONF/scripts`  folder
1. Rule modules where you have code
1. Rules consisting in a single "script" action module

Thanks to https://github.com/openhab/openhab-core/pull/3156 1. & 3. are now pretty much identical from a UI point-of-view, similar to file-defined vs. managed items or things.

> Were I king I'd change the name of Scripts to something else , maybe "Rule Fragment".

I'd propose maybe "inline scripts", like this:
- Run Script => Inline Script
- Other Rules => Rules, Scenes & Scripts

![image](https://github.com/openhab/openhab-webui/assets/2004147/10351512-5734-4ef0-b2cf-d4c198ec2afc)

It would perhaps also help to update the titles of the script modules in the core, like:
- "execute a given script" => "execute [or run] an inline script"
- "a given script evaluates to true" => "an inline script evaluates to true"

![image](https://github.com/openhab/openhab-webui/assets/2004147/dc1c0a44-48ee-4925-864f-8912cc7fe49c)

Also maybe add "scenes & scripts" to "rules" in the texts here (and associated core modules), although we don't want too much text as it could overflow.

![image](https://github.com/openhab/openhab-webui/assets/2004147/60772c30-583f-4151-acb3-226707a50573)

Maybe just "run" or "enable or disable" as the options labels then "Rules, Scenes & Scripts" as the rule picker title.

I think it's pretty clear that scheduled rules are rules, so no need to reference them here.

**Comment by timo12357** [comment]
I am not sure either of the envisioned solutions solve the problem, i.e. that scripts developed under the scripts function in the UI are not available to be used in rules.

**Comment by rkoshak** [comment]
> That's been on my mind for years, there's just too many things we refer as "scripts".
Don't forget Rules DSL Scripts and shell scripts too. It's way overloaded but every time I've spend brain cells trying to come up with a better name I've failed. If we could just come up with a meaningful name.

I wasn't aware of #3156. I may need to go amend a comment I made on a different issue.

> I'd propose maybe "inline scripts", like this:

I like that. It clears up that this part does not reference anything else and will be defined in this rule. I think in the docs we make a point to refer to them as "script actions". Maybe that can make it more clear that these are not the Scripts under Settings -> Scripts.

> "execute a given script" => "execute [or run] an inline script"

We use "run" everywhere else so I would choose run here (see "run rules" a little higher in the list in that screenshot).

> Also maybe add "scenes & scripts" to "rules" in the texts here

👍 

I agree we can skip mentioning Schedules here but I wouldn't exclude them from the list of options.

In referring to the rule picker, would it be possible to sort them rules by type? For example put Scenes at the top, followed by Scripts, then Rules and finally Schedule? I think we should not exclude Schedule because I can see use cases where there's a schedule rule that also needs to be triggered by other means. That's probably something for another issue.

> I am not sure either of the envisioned solutions solve the problem, i.e. that scripts developed under the scripts function in the UI are not available to be used in rules.

The scripts developed under Settings -> Scripts are not and have always been available for use in rules. The root problem of this issue isn't that they are not available, it's that you didn't know where to look. The suggestions I and @ghys are discussing here is to make it more clear where to look.

In short, if you want to run a Script rule from Settings -> Scripts from another rule today, you would choose "Other Rules", and from the dialog that pops up select "run these rule(s)" and select the Script from the list of *all* your rules you will see when you click on the "Rules >" line in the middle of the dialog.
